cli/bench_command.go

```diff
@@ -66,305 +62,540 @@ type benchCmd struct {
+	addPubFlags(jspub)
+	addJSCommonFlags(jspub)
+	addJSPubFlags(jspub)
+
+	jsOrdered := jsCommand.Command("ordered", "Consume JetStream messages from a consumer using an ephemeral ordered consumer").Action(c.jsOrderedAction)
+	jsOrdered.Flag("consumer", "Specify the durable consumer name to use").Default(DefaultDurableConsumerName).StringVar(&c.consumerName)
+	jsOrdered.Flag("sleep", "Sleep for the specified interval before returning from the message handler").Default("0s").DurationVar(&c.sleep)
+	addCommonFlags(jsOrdered)
+	addJSCommonFlags(jsOrdered)
+
+	jsConsume := jsCommand.Command("consume", "Consume JetStream messages from a durable consumer using a callback").Action(c.jsConsumeAction)
+	jsConsume.Flag("consumer", "Specify the durable consumer name to use").Default(DefaultDurableConsumerName).StringVar(&c.consumerName)
+	jsConsume.Flag("sleep", "Sleep for the specified interval before acknowledging the message").Default("0s").DurationVar(&c.sleep)
+	jsConsume.Flag("batch", "Sets the max number of messages that can be buffered in the client").Default("500").IntVar(&c.batchSize)
+	jsConsume.Flag("ack", "Acknowledgement mode for the consumer").Default(AckModeExplicit).EnumVar(&c.ackMode, AckModeExplicit, AckModeNone, AckModeAll)
+	addCommonFlags(jsConsume)
+	addJSCommonFlags(jsConsume)
+
+	jsFetch := jsCommand.Command("fetch", "Consume JetStream messages from a durable consumer using fetch").Action(c.jsFetchAction)
+	jsFetch.Flag("consumer", "Specify the durable consumer name to use").Default(DefaultDurableConsumerName).StringVar(&c.consumerName)
+	jsFetch.Flag("sleep", "Sleep for the specified interval before acknowledging the message").Default("0s").DurationVar(&c.sleep)
+	jsFetch.Flag("batch", "Sets the fetch batch size").Default("500").IntVar(&c.batchSize)
+	jsFetch.Flag("ack", "Acknowledgement mode for the consumer").Default(AckModeExplicit).EnumVar(&c.ackMode, AckModeExplicit, AckModeNone, AckModeAll)
+	addCommonFlags(jsFetch)
+	addJSCommonFlags(jsFetch)
+
+	kvCommand := benchCommand.Command("kv", "KV benchmark operations")
+
+	kvput := kvCommand.Command("put", "Put messages in a KV bucket").Action(c.kvPutAction)
+	kvput.Arg("bucket", "The bucket to use for the benchmark").Default(DefaultBucketName).StringVar(&c.bucketName)
+	kvput.Flag("sleep", "Sleep for the specified interval after putting each message").Default("0s").DurationVar(&c.sleep)
+	// TODO: support randomized payload data
+	addCommonFlags(kvput)
+	addJSCommonFlags(kvput)
+	addKVFlags(kvput)
+
+	kvget := kvCommand.Command("get", "Get messages from a KV bucket").Action(c.kvGetAction)
+	kvget.Arg("bucket", "The bucket to use for the benchmark").Default(DefaultBucketName).StringVar(&c.bucketName)
+	kvget.Flag("sleep", "Sleep for the specified interval after getting each message").Default("0s").DurationVar(&c.sleep)
+	addCommonFlags(kvget)
+	addJSCommonFlags(kvget)
+
+	oldJSCommand := benchCommand.Command("oldjs", "JetStream benchmark commands using the old JS API")
```

Comment:
I think we can just remove this now

Reply:
I thought about it and I did this oldjs because I thought it would be useful to compare quickly the difference in speed (if any) between the old and the new JS API. So my vote is to keep it there for the moment and could always remove it later.

As a half-way compromise what I just did is mark the 'oldjs' command as hidden, so it's still there but you won't see it :).